### PR TITLE
feat: tailcall added

### PR DESCRIPTION
**Pull request by Dhanus3133**
Added [Tailcall](https://github.com/tailcallhq/tailcall/)

### CODE REVIEW

File: tailcall/tailcall.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+wget https://github.com/tailcallhq/tailcall/releases/download/v0.13.0/tailcall-x86_64-unknown-linux-gnu
```
Comment (sc231997) [inline tailcall/tailcall.sh:2]: using hardcoded version is not recommended. Rather use following:
`
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
brew tap tailcallhq/tailcall
brew install tailcall
tc ./hello.graphql
`

Comment (Dhanus3133) [reply tailcall/tailcall.sh]: Thanks, modified it!

File: tailcall/hello.graphql [hunk 2]
```diff
@@ -0,0 +1,7 @@
+schema @server(port: 8000, enableHttpCache: true, enableGraphiql: true) {
```
Comment (tusharmath) [inline tailcall/hello.graphql:1]:
```suggestion
schema @server(port: 8000) {
```

**Comment by tusharmath** [comment]
Where can we see the benchmark results?

**Comment by sc231997** [comment]
> Where can we see the benchmark results?

image
https://github.com/graphql-crystal/benchmarks/actions/runs/6736714074/job/18312641720?pr=210

File: benchmarks.yaml [hunk 3]
```diff
@@ -200,4 +200,12 @@
   server: HonoJS
   run:
     cmd: bun
-    args: ["index.js"]
\ No newline at end of file
+    args: ["index.js"]
+- id: tailcall
+  name: tailcall
+  url: https://github.com/tailcallhq/tailcall
+  lang: Rust
```
Comment (tusharmath) [inline benchmarks.yaml:207]: No Code

Comment (sc231997) [reply benchmarks.yaml:207]: For `lang`?

Comment (tusharmath) [reply benchmarks.yaml:207]: Yes. Because the dev is actually not writing any rust code right?

Comment (sc231997) [reply benchmarks.yaml:207]: Yes that's correct. But not sure what does `lang` indicates here...implementation language or in which client will be using it.

File: tailcall/hello.graphql (hunk 2 above)
Comment (Dhanus3133) [reply tailcall/hello.graphql]: Thanks, modified that!

File: benchmarks.yaml (hunk 3 above)
Comment (sc231997) [reply benchmarks.yaml]: Readme.md files indicate `lang` as server language here. So Rust is correct

Comment (Dhanus3133) [reply benchmarks.yaml]: @tusharmath Done!

Comment (Dhanus3133) [reply benchmarks.yaml]: @sc231997  This is why you must at least understand before blind copy-pasting someone's code.

File: benchmarks.yaml [hunk 4]
```diff
@@ -95,23 +95,6 @@
...
-    args:
-      [
-        "-w",
-        <%= System.cpu_count.to_s %>,
-        "-s",
-        "-t",
-        "2",
-        "-b",
-        "tcp://127.0.0.1:8000",
-      ]
```
Comment (tusharmath) [inline benchmarks.yaml:114]: Why should we disable graphql-ruby?

File: benchmarks.yaml [hunk 5]
```diff
@@ -143,20 +126,20 @@
...
-  server: Akka HTTP
-  run:
-    cmd: java
-    args:
-      [
-        "-Xrs",
-        "-Xmx4G",
-        "-jar",
-        "./target/scala-2.13/sangria-assembly-0.1.0-SNAPSHOT.jar",
-      ]
```
Comment (tusharmath) [inline benchmarks.yaml:159]: Why should we disable sangria?

File: benchmarks.yaml [hunk 6]
```diff
@@ -200,4 +183,28 @@
...
+    args:
+      [
+        "-w",
+        <%= System.cpu_count.to_s %>,
+        "-s",
+        "-t",
+        "2",
+        "-b",
+        "tcp://127.0.0.1:8000",
+      ]
```
Comment (tusharmath) [inline benchmarks.yaml:203]: Why re-order the servers?

File: benchmarks.yaml (hunk 5 above)
Comment (Dhanus3133) [reply benchmarks.yaml:159]: [Action](https://github.com/graphql-crystal/benchmarks/actions/runs/6730873449/job/18294413019#step:9:502)

Check this one, it just halts there on sangria

File: benchmarks.yaml (hunk 4 above)
Comment (Dhanus3133) [reply benchmarks.yaml:114]: I didn't disable but reordered it

File: benchmarks.yaml (hunk 6 above)
Comment (Dhanus3133) [reply benchmarks.yaml:203]: Check this one [Action](https://github.com/graphql-crystal/benchmarks/actions/runs/6630650384/job/18012480495#step:9:481)

Also quite some more PR's have been failed do you this
```
/home/runner/work/benchmarks/benchmarks/graphql-ruby: command 'bundle' failed with exit code 1
```

So I reordered it

Comment (Dhanus3133) [reply benchmarks.yaml:203]: But I'm not quite sure, that reordering can fix that problem all time.

File: benchmarks.yaml (hunk 5 above)
Comment (tusharmath) [reply benchmarks.yaml:159]: @jgillich could comment on it.

**Comment by tusharmath** [comment]
For reference here is the generated README.md


### Results

| Name                          | Language      | Server          | Latency avg      | Requests      |
| ----------------------------  | ------------- | --------------- | ---------------- | ------------- |
| [static-rust](https://actix.rs/) | Rust | Actix Web | 1.22ms | 82kps |
| [tailcall](https://github.com/tailcallhq/tailcall) | No Code | Tailcall | 2.33ms | 43kps |
| [graphql-crystal](https://github.com/graphql-crystal/graphql) | Crystal | Kemal | 2.83ms | 35kps |
| [Juniper](https://github.com/graphql-rust/juniper) | Rust | Actix Web | 2.90ms | 34kps |
| [async-graphql](https://github.com/async-graphql/async-graphql) | Rust | Actix Web | 2.96ms | 34kps |
| [gqlgen](https://github.com/99designs/gqlgen) | Go | net/http | 3.26ms | 31kps |
| [Mercurius](https://github.com/mercurius-js/mercurius) | Node.js | Fastify | 5.93ms | 17kps |
| [Hot Chocolate](https://github.com/ChilliCream/hotchocolate) | C# | ASP.NET | 7.02ms | 14kps |
| [agoo](https://github.com/ohler55/agoo) | Ruby/C | agoo | 7.92ms | 13kps |
| [graphql-go](https://github.com/graphql-go/graphql) | Go | net/http | 8.47ms | 12kps |
| [graphql-yoga](https://github.com/dotansimha/graphql-yoga) | Node.js | http | 11.76ms | 8.5kps |
| [Hono](https://github.com/honojs/graphql-server) | Bun | HonoJS | 13.05ms | 7.7kps |
| [graphql-jit](https://github.com/zalando-incubator/graphql-jit) | Node.js | http | 17.44ms | 5.7kps |
| [Absinthe](https://github.com/absinthe-graphql/absinthe) | Elixir | Phoenix | 18.30ms | 5.5kps |
| [apollo](https://github.com/apollographql/apollo-server) | Node.js | Express | 24.41ms | 4.1kps |
| [graphql-ruby](https://github.com/rmosolgo/graphql-ruby) | Ruby | Puma | 39.82ms | 3.2kps |
| [graphql-js](https://github.com/graphql/graphql-js) | Node.js | http | 34.32ms | 2.9kps |
| [Strawberry](https://github.com/strawberry-graphql/strawberry) | Python | gunicorn | 83.02ms | 1.2kps |
| [Graphene](https://github.com/graphql-python/graphene) | Python | gunicorn | 83.47ms | 1.2kps |

File: benchmarks.yaml [hunk 7]
```diff
@@ -200,4 +183,28 @@
...
+- id: tailcall
+  name: tailcall
+  url: https://github.com/tailcallhq/tailcall
+  lang: No Code
```
Comment (jgillich) [inline benchmarks.yaml:207]: `Rust`

**Comment by jgillich** [comment]
Hmm, the @tailcallhq/tailcall package does nothing for me. And when I try to install it directly, I get an error:

```
$ npm i @tailcallhq/core-linux-x86_64-gnu
npm ERR! code EBADPLATFORM
npm ERR! notsup Unsupported platform for @tailcallhq/core-linux-x86_64-gnu@0.14.14: wanted {"os":"linux-gnu","cpu":"x86_64"} (current: {"os":"linux","cpu":"x64"})
npm ERR! notsup Valid os:   linux-gnu
npm ERR! notsup Actual os:  linux
npm ERR! notsup Valid cpu:  x86_64
npm ERR! notsup Actual cpu: x64
```

`linux-gnu` and `linux-musl` aren't a thing according to https://nodejs.org/api/process.html#processplatform

**Pull request closed without merging**